plugins/producer_plugin/producer_plugin.cpp

```diff
@@ -3050,17 +3052,23 @@ void producer_plugin::received_block(uint32_t block_num, chain::fork_db_add_t fo
    my->_received_block = block_num;
    // fork_db_add_t::fork_switch means head block of best fork (different from the current branch) is received.
    // Since a better fork is available, interrupt current block validation and allow a fork switch to the better branch.
-   if (fork_db_add_result == fork_db_add_t::fork_switch) {
-      fc_ilog(_log, "new best fork received");
-      my->chain_plug->chain().interrupt_apply_block_transaction();
+   if (my->_is_savanna_active) { // interrupt during transition causes issues, so only allow after transition
+      if (fork_db_add_result == fork_db_add_t::appended_to_head) {
+         fc_tlog(_log, "new head block received");
```

Comment:
Would it make sense to have this message?
```suggestion
         fc_tlog(_log, "new head block received, interrupting transaction");
```

Reply:
https://github.com/AntelopeIO/spring/pull/1330/commits/ba2c739f0a0ed32c804b38fe5fcfadaba63a4a4c